File: DS_LAB/Tree/BST/levelOrder.py
from collections import deque


class Node:
    def __init__(self, data):
        self.data = data
        self.left = None
        self.right = None


class BST:

    def __init__(self):
        self.root = None

    def insertNode(self, data):
        if self.root is None:
            self.root = Node(data)
        else:
            self._insertHelper(self.root, data)

    def _insertHelper(self, currNode, data):

        if data < currNode.data:
            if currNode.left:
                self._insertHelper(currNode.left, data)
            else:
                currNode.left = Node(data)
        else:
            if currNode.right:
                self._insertHelper(currNode.right, data)
            else:
                currNode.right = Node(data)

    def levelOrderTraversal(self):
        if self.root is None:
            return "Tree is Empty"

        q = deque()
        q.append(self.root)

        while q:
            curr = q.popleft()
            print(curr.data)
            if curr.left:
                q.append(curr.left)
            if curr.right:
                q.append(curr.right)


bst = BST()
bst.insertNode("F")
bst.insertNode("B")
bst.insertNode("G")
bst.insertNode("A")
bst.insertNode("D")

bst.levelOrderTraversal()

File: DS_LAB/Tree/BST/level_order.c

#include <stdio.h>
#include <stdlib.h>
#include <ctype.h>

typedef struct node
{
    char data;
    struct node *left;
    struct node *right;
} node;

node *createNode(int data)
{
    node *newNode = (node *)malloc(sizeof(node));
    newNode->data = data;
    newNode->left = NULL;
    newNode->right = NULL;

    return newNode;
}

node *insertNode(node *root, int data)
{
    if (root == NULL)
    {
        root = createNode(data);
        return root;
    }
    else if (data <= root->data)
        root->left = insertNode(root->left, data);
    else
        root->right = insertNode(root->right, data);

    return root;
}

int main()
{
    node *root = NULL;
    root = Insert(root, 'M');
    root = Insert(root, 'B');
    root = Insert(root, 'Q');
    root = Insert(root, 'Z');
    root = Insert(root, 'A');
    root = Insert(root, 'C');
}